doc/spec/manifest.md

```diff
@@ -0,0 +1,218 @@
+# Distribution Content Manifests
+
+A Content Manifest is a simple JSON file which contains general fields that
+are typical to any package management system. The goal is for these manifests
+to describe an application and its dependencies in a content-addressable and
+verifiable way.
+
+## Field Descriptions
+
+All fields are required unless marked `(optional)`.
+
+- **`schemaVersion`** *int*
+
+    Specifies the schema version of the manifest as an integer. This document 
+    describes version `2` only.
+
+- **`createdAt`** *string*
+
+    ISO-8601 formatted combined date and time at which the manifest was
+    created.
+
+- **`maintainers`** *array*
+
+    An array of objects representing persons or organizations which are
+    designated as mainainers of the described content. Fields are:
+    
+    - **`name`** *string*
+    
+        The name of a maintaining person or organization.
+    
+    - **`email`** *string* `(optional)`
+    
+        The email address of a maintainer.
+    
+    - **`url`** *string* `(optional)`
+    
+        A URL to a maintainer's website, blog, etc.
+
+- **`description`** *string*
+
+    A brief description of the content or application.
+
+- **`keywords`** *array*
+
+    An array of strings which should be considered keywords that may be used
+    by a separate search system to index content.
+
+- **`website`** *string*
+
+    A URL to a related website or project page for the application.
+
+- **`licenses`** *array*
+
+    An array of strings which identify any software licenses which apply
+    to the application. Values should be a [SPDX License Identifier](https://spdx.org/licenses/).
+
+- **`dependencies`** *array*
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`length`** *int*
+    
+        The length in bytes of the dependency object.
+    
+    - **`digest`** *string*
+    
+        A string labeled with a digest algorithm prefix and the hex-encoded
+        value of that digest for the content.
+
+        Digest strings should be of the form:
+
+            <digest-alg-name>:<hex-encoded-digest>
```

Comment:
It could be helpful to specify `digest-alg-name`'s regex. In `sha256:deadbeefdeadbeef...@sha512:<hex-encoded-digest>`, with the current spec, `digest-alg-name` could very well be everything up until sha512 and thus create a (admittedly crazy) ambiguity with the <repository> form. You might as well specify `<repository>` too.